docs/lang_spec/Cadence Language Specification.md

```diff
@@ -0,0 +1,5798 @@
+
+Structs in Cadence are declared using the struct keyword, followed by an identifier (the name of the struct), optional conformances to interfaces, and a block containing the struct’s fields and any associated functions. The access level of a struct and its members is controlled by access modifiers, allowing developers to determine the visibility and accessibility of struct elements.
+
+- **Syntax**:
+  - The basic syntax for a struct declaration is as follows:
+    ```cadence
+    access(struct) struct StructName {
+      access(struct) let constantField: Type
+      access(struct) var variableField: Type
+      
+      init(parameterName: Type) {
+        // Initialization of fields
+        self.constantField = parameterValue
+        self.variableField = anotherValue
+      }
+      
+      fun someFunction() {
+        // Function body
+      }
+    }
+
+    ```
+    - **access**: Specifies the access control for the struct and its members (e.g., pub, access(contract), access(self)).
+    - **StructName**: The name of the struct.
+    - **Fields**: Declared with the let or var keywords. Constant fields (let) are immutable after initialization, while variable fields (var) can be modified after the struct is created.
+    - **Initializer**: The init function is responsible for initializing all fields in the struct. It ensures that each field is assigned a value before the struct is used.
+    - **Functions**: Structs can contain functions, which operate on their fields and are defined within the struct body.
+
+- **Example**:
+  ```cadence
+  pub struct User {
+    pub let id: Int
+    pub var name: String
+    
+    init(id: Int, name: String) {
+      self.id = id
+      self.name = name
+    }
+     
+    pub fun updateName(newName: String) {
+      self.name = newName
+    }
+  }
+  ```
+  - In this example, the User struct has a constant field id and a variable field name. The initializer sets both fields, and the updateName function allows modification of the name field after the struct is created.
+
+- **EBNF**:
+  - The formal EBNF for struct declarations in Cadence is as follows:
+    ```ebnf
+    structDeclaration
+      : access 'struct' identifier conformances? '{' membersAndNestedDeclarations '}'
+      ;
+    
+    membersAndNestedDeclarations
+      : (fieldDeclaration | functionDeclaration | structDeclaration)*
+      ;
+  ```
+    - **access**: The access control for the struct (e.g., pub for public, priv for private).
+    - **identifier**: The name of the struct.
+    - **conformances**: Optional interfaces the struct conforms to.
+    - **membersAndNestedDeclarations**: Defines the fields, functions, and nested structs within the struct.
```

Comment:
More of a general proposal/suggestion on the grammar formatting: I think we could make those non-terminals in a production-rule to link to its definition. e.g: link `access` in the above production-rule to the section where access control's grammar is defined. Something [like this](https://gist.github.com/SupunS/3efb34060cf6cc43fa292e33cd51c839).
Unfortunately, this is not easy to get it working, would most probably need some post-processor to generate links, do formatting, etc.

But in the meantime, maybe extract those to separate sections (e.g: `access`, `conformance`, etc.), and remove it from here (can link later), as those can get repetitive.